app/utils/documents.js

```diff
@@ -87,6 +115,38 @@ export const restorePiecesFromPreviousAgendaitem = async function(agendaitem, do
     body: JSON.stringify(body),
   });
   if (!response.ok) {
-    throw new Error(`Failed to restore pieces from container ${documentContainer.get('id')} to agendaitem ${agendaitem.get('id')}`);
+    throw new Error(
+      `Failed to restore pieces from container ${documentContainer.get(
+        'id'
+      )} to agendaitem ${agendaitem.get('id')}`
+    );
   }
 };
+
+/**
+ * takes resolved file relationship
+ * reason: in order to avoid async getters: Ember does not handle these well
+ */
+export function getPieceDisplayName(piece, file) {
+  if (file) {
+    return `${piece.name}.${file.extension}`;
+  } else {
+    return piece.name;
+  }
+}
+
+/**
+ * takes resolved file relationship
+ * reason: in order to avoid async getters: Ember does not handle these well
+ */
+export function getPieceDownloadUrl(piece, file) {
+  if (file) {
+    const pieceDisplayName = getPieceDisplayName(piece, file);
+    const downloadFilename = sanitize(pieceDisplayName, {
+      replacement: '_',
+    });
+    return `${file.downloadLink}?name=${encodeURIComponent(downloadFilename)}`;
+  } else {
+    return undefined;
+  }
+}
```

Comment:
aangezien je een return hebt in de `if` blokken, is de else definitie niet nodig.
```
if (iets) {
return;
}
return;
```